app-backend/db/src/main/scala/com/azavea/rf/database/notification/templates/PlainGroupRequest.scala

```diff
@@ -0,0 +1,39 @@
+package com.azavea.rf.database.notification.templates
+
+import com.azavea.rf.database._
+import com.azavea.rf.datamodel._
+
+import doobie.ConnectionIO
+
+import java.util.UUID
+
+case class PlainGroupRequest(
+  groupId: UUID,
+  groupType: GroupType,
+  subjectId: String
+) {
+  def build: ConnectionIO[String] = {
+    for {
+      subjectEmail <- UserDao.unsafeGetUserById(subjectId).map((usr: User) => usr.email)
+      groupName <- groupType match {
+        case GroupType.Team =>
+          TeamDao.unsafeGetTeamById(groupId) map { (team: Team) => team.name }
+        case GroupType.Organization =>
+          OrganizationDao.unsafeGetOrganizationById(groupId) map {
+            (org: Organization) => org.name
+          }
+        case _ =>
+          throw new IllegalArgumentException(
+            "Users can only request to add themselves to organizations and teams"
+          )
+      }
+    } yield {
```

Comment:
I think there's an open question here for where an organization admin will accept/deny user requests in the UI. The closest we have is the UI for viewing members of an organization. Knowing where this will live in the UI will let us put a parameterized link in the email.

@designmatty do you have any suggestions?

Reply:
I shared this with Alex already, but to be thorough, this is the updated interface which shows this interaction https://app.goabstract.com/projects/cd34e180-d61d-11e7-b99b-5793a477ef34/branches/95c06d80-761f-11e8-bf41-df47385a2ce8/commits/32707e37862e20733287063a90de5b3b5ebed50c/files/B7A9D19F-97DB-441C-97ED-F692000DA14B/layers/38B7CDBB-650A-4404-8DFB-9932B7CA7FF5